File: java/Film.java
public class Film extends Tayangan {
    // atribut
    private String genre;
    private String sutradara;
    private String ratingUsia;
    private String bahasa;
    private int tahunRilis;


    // konstruktor kosong
    public Film(){
    }

    // konstruktor
    public Film(String id, String judul, int durasi, String jamTayang, double hargaTiket, 
        String genre, String sutradara, String ratingUsia, String bahasa, int tahunRilis){
        super(id, judul, durasi, jamTayang, hargaTiket); 
        this.genre = genre;
        this.sutradara = sutradara;
        this.ratingUsia = ratingUsia;
        this.bahasa = bahasa;
        this.tahunRilis = tahunRilis;
    }

    // getter
    public String getGenre(){
        return genre;
    }
    public String getSutradara(){
        return sutradara;
    }
    public String getRatingUsia(){
        return ratingUsia;
    }
    public String getBahasa(){
        return bahasa;
    }
    public int getTahunRilis(){
        return tahunRilis;
    }

    // setter
    public void setGenre(String genre){
        this.genre = genre;
    }
    public void setSutradara(String sutradara){
        this.sutradara = sutradara;
    }
    public void setRatingUsia(String ratingUsia){
        this.ratingUsia = ratingUsia;
    }
    public void setBahasa(String bahasa){
        this.bahasa = bahasa;
    }
    public void setTahunRilis(int tahunRilis){
        this.tahunRilis = tahunRilis;
    }
}

File: cpp/Film.cpp
#include <iostream>
#include <string>
#include "Tayangan.cpp"
using namespace std;

class Film : public Tayangan{
private:
    // atribut
    string genre;
    string sutradara;
    string ratingUsia;
    string bahasa;
    int tahunRilis;

public:
    // konstruktor kosong
    Film(){
    }

    // konstruktor
    Film(string id, string judul, int durasi, string jamTayang, double hargaTiket, 
        string genre, string sutradara, string ratingUsia, string bahasa, int tahunRilis) : 
        Tayangan (id, judul, durasi, jamTayang, hargaTiket){
        this->genre = genre;
        this->sutradara = sutradara;
        this->ratingUsia = ratingUsia;
        this->bahasa = bahasa;
        this->tahunRilis = tahunRilis;
    }

    // getter 
    string getGenre(){
        return genre;
    }
    string getSutradara(){
        return sutradara;
    }
    string getRatingUsia(){
        return ratingUsia;
    }
    string getBahasa(){
        return bahasa;
    }
    int getTahunRilis(){
        return tahunRilis;
    }
    // setter
    void setGenre(string genre){
        this->genre = genre;
    }
    void setSutradara(string sutradara){
        this->sutradara = sutradara;
    }
    void setRatingUsia(string ratingUsia){
        this->ratingUsia = ratingUsia;
    }
    void setBahasa(string bahasa){
        this->bahasa = bahasa;
    }
    void setTahunRilis(int tahunRilis){
        this->tahunRilis = tahunRilis;
    }

    // destruktor
    ~Film(){
    }
};